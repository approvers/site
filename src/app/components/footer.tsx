"use client";

import { Box, Link } from "@chakra-ui/react";

import NextLink from "next/link";
import { useContext } from "react";
import { yearContext } from "../lib/year-context";

export const Footer = (): JSX.Element => {
  const year = useContext(yearContext);
  return (
    <Box as="footer" p={4} textAlign="center" borderColor="shadowed" borderTopWidth="2px">
      <Link as={NextLink} mr={2} color="highlighted" fontStyle="normal" href="/policy">
        プライバシーポリシー
      </Link>
      - (c) {year} Approvers
    </Box>
  );
};
