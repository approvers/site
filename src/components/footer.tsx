import { Center, Link } from "@chakra-ui/react";
import NextLink from "next/link";

export const Footer = (): JSX.Element => (
  <Center w="100%" p={4} borderTop="2px" borderColor="shadowed">
    <NextLink href="/policy" passHref>
      <Link mr={2} fontStyle="normal" color="highlighted">
        プライバシーポリシー
      </Link>
    </NextLink>
    - (c) 2021 Approvers
  </Center>
);
