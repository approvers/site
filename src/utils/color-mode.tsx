"use client";

import type { IconButtonProps, SpanProps } from "@chakra-ui/react";
import { ClientOnly, IconButton, Skeleton, Span } from "@chakra-ui/react";
import { ThemeProvider, useTheme } from "next-themes";
import type { ThemeProviderProps } from "next-themes";
import * as React from "react";
import { LuMoon, LuSun } from "react-icons/lu";

export interface ColorModeProviderProps extends ThemeProviderProps {}

export function ColorModeProvider(props: ColorModeProviderProps) {
  return <ThemeProvider attribute="class" disableTransitionOnChange {...props} />;
}

export type ColorMode = "light" | "dark";

export interface UseColorModeReturn {
  colorMode: ColorMode;
  setColorMode: (colorMode: ColorMode) => void;
  toggleColorMode: () => void;
}

export function useColorMode(): UseColorModeReturn {
  const { resolvedTheme, setTheme, forcedTheme } = useTheme();
  const colorMode = forcedTheme || resolvedTheme;
  const toggleColorMode = () => {
    setTheme(resolvedTheme === "dark" ? "light" : "dark");
  };
  return {
    colorMode: colorMode as ColorMode,
    setColorMode: setTheme,
    toggleColorMode,
  };
}

export function useColorModeValue<T>(light: T, dark: T) {
  const { colorMode } = useColorMode();
  return colorMode === "dark" ? dark : light;
}

export function ColorModeIcon() {
  const { colorMode } = useColorMode();
  return colorMode === "dark" ? <LuMoon /> : <LuSun />;
}

export function OppositeColorModeIcon() {
  const { colorMode } = useColorMode();
  return colorMode === "dark" ? <LuSun /> : <LuMoon />;
}

interface ColorModeButtonProps extends Omit<IconButtonProps, "aria-label"> {}

export const ColorModeButton = React.forwardRef<HTMLButtonElement, ColorModeButtonProps>(
  function ColorModeButton(props, ref) {
    const { toggleColorMode } = useColorMode();
    return (
      <ClientOnly fallback={<Skeleton boxSize="9" />}>
        <IconButton
          ref={ref}
          aria-label="Toggle color mode"
          onClick={toggleColorMode}
          size="sm"
          variant="ghost"
          {...props}
          css={{
            _icon: {
              width: "5",
              height: "5",
            },
          }}
        >
          <ColorModeIcon />
        </IconButton>
      </ClientOnly>
    );
  },
);

export const LightMode = React.forwardRef<HTMLSpanElement, SpanProps>(
  function LightMode(props, ref) {
    return (
      <Span
        className="chakra-theme light"
        ref={ref}
        display="contents"
        color="fg"
        colorPalette="gray"
        colorScheme="light"
        {...props}
      />
    );
  },
);

export const DarkMode = React.forwardRef<HTMLSpanElement, SpanProps>(function DarkMode(props, ref) {
  return (
    <Span
      className="chakra-theme dark"
      ref={ref}
      display="contents"
      color="fg"
      colorPalette="gray"
      colorScheme="dark"
      {...props}
    />
  );
});
