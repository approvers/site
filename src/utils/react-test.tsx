import reactTestRenderer, { ReactTestRenderer, TestRendererOptions } from "react-test-renderer";
import { ReactElement } from "react";

export const render = (
  ui: ReactElement,
  options?: TestRendererOptions | undefined,
): ReactTestRenderer => reactTestRenderer.create(ui, options);
