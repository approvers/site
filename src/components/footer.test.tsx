import { expect, it } from "vitest";
import { Footer } from "./footer";
import { render } from "../utils/react-test";

it("renders correctly", () => {
  const tree = render(<Footer />);
  expect(tree.toJSON()).toMatchSnapshot();
});
