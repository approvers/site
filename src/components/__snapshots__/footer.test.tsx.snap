// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-1n9yuqd"
>
  <a
    className="chakra-link css-wppl45"
    href="/policy"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    プライバシーポリシー
  </a>
  - (c) 2021 Approvers
</div>
`;
